src/java.base/share/classes/java/net/URL.java

```diff
@@ -1384,9 +1384,13 @@ public URLStreamHandler run() {
      * Returns the protocol in lower case. Special cases known protocols
      * to avoid loading locale classes during startup.
      */
-    static String toLowerCase(String protocol) {
-        if (protocol.equals("jrt") || protocol.equals("file") || protocol.equals("jar")) {
-            return protocol;
+    static String lowerCaseProtocol(String protocol) {
+        if (protocol.equals("jrt")) {
```

Comment:
I don't get the difference between chaining with `||` and `if-else if`?

Reply:
There'd be no point in changing from a `||`-chain to `if.. else if` unless there's also a need to have different results happen from either case. Here we now want to return the appropriate constant (`"jrt"` etc), thus splitting the `||` is necessary.